frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,675 @@
+  React.useEffect(() => {
+    if (state.status === FormStatus.Submitting) {
+      async function sendMessage() {
+        const input: CreateFormMessageInput = {
+          data: {
+            name: state.values.name,
+            email: state.values.email,
+            message: state.values.message,
+          },
+        };
+        const response = await fetchGraphQL<
+          InsertFormMessageMutation,
+          InsertFormMessageMutationVariables
+        >(InsertFormMessage, { input });
+
+        if (response.createFormMessage == null) {
+          dispatch({ type: FormActionType.Failed });
+        } else {
+          dispatch({ type: FormActionType.Success });
+        }
+      }
+      sendMessage();
+    }
+  }, [state.status, state.values]);
+
+  return (
+    <Box
+      flexWrap="wrap"
+      fontFamily="Roboto Mono"
+      id="contact"
+      mt={{
+        base: "20",
+        md: "0",
+      }}
+      mr={{
+        base: "-2.5",
+        md: "0",
+      }}
+      display="flex"
+      flexGrow={1}
+      pb="20"
+      flexDir={{
+        base: "column",
+        sm: "row",
+        md: "column",
+        lg: "row",
+      }}>
+      <Box
+        display={state.status == FormStatus.Submitted ? "inline-block" : "none"}
+        pb="12">
+        <Text
+          as="h3"
+          fontSize={{ base: "3xl", sm: "4xl", md: "5xl" }}
+          fontWeight="bold"
+          fontFamily="Europa"
+          lineHeight="1.1em"
+          pb="5">
+          Thank you!
+        </Text>
+        <Text as="p" fontSize="sm" color="#5c5c5c">
+          We&apos;ll get in touch soon.
+        </Text>
+      </Box>
+      <Box
+        noValidate
+        onSubmit={() => dispatch({ type: FormActionType.Submit })}
+        as="form"
+        display={
+          state.status == FormStatus.Submitted ? "none" : "inline-block"
+        }>
+        <Box
+          w={{
+            base: "calc(100% - 20px)",
+            sm: "calc(50% - 20px)",
+            md: "calc(100% - 20px)",
+            lg: "calc(50% - 20px)",
+          }}
+          m="2.5"
+          display="inline-block"
+          pos="relative">
+          <FormControl
+            id="name"
+            isInvalid={
+              state.validationErrors.name == "" ||
+              state.validationErrors.name == null
+                ? false
+                : true
+            }>
+            <FormLabel
+              fontWeight="400"
+              fontSize="sm"
+              textTransform="uppercase"
+              letterSpacing="0.1em"
+              pos="relative"
+              w="full"
+              display="block"
+              mb="0">
+              Name
+            </FormLabel>
+            <Input
+              onChange={(event) => {
+                dispatch({
+                  type: FormActionType.UpdateField,
+                  value: event.target.value,
+                  name: "name",
+                });
+              }}
```

Comment:
With a little trick, this could be extracted into a reusable event handler, that can be used by all three inputs. The trick is to set the attribute `name` on the Input, so in this case it would be `name="name"`, for email it would be `name="email"`, etc. Then you create an event handler like this:

```ts
const onFieldChange = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
  const  { name, value } = event.target;
  dispatch({
    type: FormActionType.UpdateField,
    name,
    value
  });
}, []);
```

This has also the benefit that the function remains stable across renders